core/transaction-pool/graph/src/pool.rs

```diff
@@ -165,6 +165,17 @@ impl<B: ChainApi> Pool<B> {
 		)
 	}
 
+	/// Watch existing transaction
+	///
+	/// Get notified when some existing transaction is finished verifying or gets finalized
+	/// in a new block.
+	pub fn watch(
+		&self,
+		hash: ExHash<B>,
+	) -> Watcher<ExHash<B>, BlockHash<B>> {
+		self.validated_pool.watch(hash)
```

Comment:
Hmm, actually just realised that exposing this might open up a DoS vector, since we don't really validate if the hash is in the pool (which for one reason is good - you can subscribe to something before you saw it), but on the other hand you may wait for it forever.

What's your thoughts on this?

Reply:
We can hard limit txpool number of watchers?